packages/fxa-auth-server/lib/routes/subscriptions/stripe.ts

```diff
@@ -416,51 +416,65 @@ export class StripeHandler {
   async previewInvoice(
     request: AuthRequest
   ): Promise<invoiceDTO.firstInvoicePreviewSchema> {
-    this.log.begin('subscriptions.previewInvoice', request);
+    try {
+      this.log.begin('subscriptions.previewInvoice', request);
+
+      const { promotionCode, priceId } = request.payload as Record<
+        string,
+        string
+      >;
+
+      let customer = undefined;
+      if (request.auth.credentials) {
+        const { uid, email } = await handleAuth(this.db, request.auth, true);
+        await this.customs.check(request, email, 'previewInvoice');
+        try {
+          customer = await this.stripeHelper.fetchCustomer(uid, ['tax']);
+        } catch (e: any) {
+          this.log.error('previewInvoice.fetchCustomer', { error: e, uid });
+        }
+      } else {
+        await this.customs.checkIpOnly(request, 'previewInvoice');
+      }
 
-    const { promotionCode, priceId } = request.payload as Record<
-      string,
-      string
-    >;
+      const taxAddress = this.buildTaxAddress(
+        request.app.clientAddress,
+        request.app.geo.location
+      );
 
-    let customer = undefined;
-    if (request.auth.credentials) {
-      const { uid, email } = await handleAuth(this.db, request.auth, true);
-      await this.customs.check(request, email, 'previewInvoice');
       try {
-        customer = await this.stripeHelper.fetchCustomer(uid, ['tax']);
-      } catch (e: any) {
-        this.log.error('previewInvoice.fetchCustomer', { error: e, uid });
-      }
-    } else {
-      await this.customs.checkIpOnly(request, 'previewInvoice');
-    }
-
-    const taxAddress = this.buildTaxAddress(
-      request.app.clientAddress,
-      request.app.geo.location
-    );
+        const previewInvoice = await this.stripeHelper.previewInvoice({
+          customer: customer || undefined,
+          promotionCode,
+          priceId,
+          taxAddress,
+        });
 
-    try {
-      const previewInvoice = await this.stripeHelper.previewInvoice({
-        customer: customer || undefined,
-        promotionCode,
-        priceId,
-        taxAddress,
+        return stripeInvoiceToFirstInvoicePreviewDTO(previewInvoice);
+      } catch (err: any) {
+        if (err.type === 'StripeInvalidRequestError') {
+          throw error.invalidInvoicePreviewRequest(
+            err,
+            err.message,
+            priceId,
+            customer?.id
+          );
+        } else {
+          throw err;
+        }
+      }
+    } catch (err: any) {
+      //TODO - this is part of 7664, we can remove this one we uncover the underlying error
```

Comment:
[non-blocking] Mind adding the FXA- prefix so it's more easily searchable?

Reply:
added!